src/index.ts

```diff
@@ -0,0 +1,115 @@
+interface IToDoList {
+  get tasks(): Task[];
+  getTask(id: TaskId): Task | null;
+  addTask(name: string, body: string, isRequireEditConfirm?: boolean): void;
+  removeTask(id: TaskId): void;
+  editTask(id: TaskId, data: { name: string; body: string }): void;
+  toogleCompleted(id: TaskId): void;
+  getTasksAmount(): number;
+  getNotCompletedAmount(): number;
+  searchTasks(value: string): Task[];
+  sortByIsCompleted(isCompletedFirst: boolean): Task[];
+  sortByCreationDate(isAscending: boolean): Task[];
```

Comment:
There is a requirement: "It is also necessary to separately expand the list with the ability to sort notes by status or time of creation."

It means both sort methods should be implemented in a separate class.